tests/test_alchemy.py

```diff
@@ -74,6 +107,56 @@ def test_pk_force_value(self):
         self.assertEqual(0, std2.id)
 
 
+class SQLAlchemyGetOrCreateTests(unittest.TestCase):
+    def setUp(self):
+        models.session.rollback()
+
+    def test_simple_call(self):
+        obj1 = WithGetOrCreateFieldFactory(foo='foo1')
+        obj2 = WithGetOrCreateFieldFactory(foo='foo1')
+        self.assertEqual(obj1, obj2)
+
+    def test_missing_arg(self):
+        with self.assertRaises(factory.FactoryError):
+            MultifieldModelFactory()
+
+    def test_multicall(self):
+        objs = MultifieldModelFactory.create_batch(
+            6,
+            slug=factory.Iterator(['main', 'alt']),
+        )
+        self.assertEqual(6, len(objs))
+        self.assertEqual(2, len(set(objs)))
+        self.assertEqual(
+            list(
+                obj.slug for obj in models.session.query(
+                    models.MultiFieldModel.slug
+                )
+            ),
+            ["alt", "main"],
+        )
+
+
+class MultipleGetOrCreateFieldsTest(unittest.TestCase):
+    def setUp(self):
+        models.session.rollback()
+
+    def test_one_defined(self):
+        obj1 = WithGetOrCreateFieldFactory()
```

Comment:
This should be using `WithMultipleGetOrCreateFieldsFactory`, it currently duplicates `test_simple_call`.

Reply:
Thank you, I found out I was missing adding non get or create fields to object on create, and also wrapping db errors inside a catch block.

This pointed me in refactoring some code into the `_save` method.

And thank you again for noticing these rookie mistakes.